alf/algorithms/muzero_representation_learner.py

```diff
@@ -55,31 +55,48 @@
 
 
 @alf.configurable
-class MuzeroAlgorithm(OffPolicyAlgorithm):
-    """MuZero algorithm.
+class MuzeroRepresentationLearner(OffPolicyAlgorithm):
+    """MuZero-style Representation Learner.
 
     MuZero is described in the paper:
     `Schrittwieser et al. Mastering Atari, Go, Chess and Shogi by Planning with a Learned Model <https://arxiv.org/abs/1911.08265>`_.
 
     The pseudocode can be downloaded from `<https://arxiv.org/src/1911.08265v2/anc/pseudocode.py>`_
+
+    This representation learner trains the underlying MCTSModel to
+
+    1) Most importantly, produce a latent representation from an observation
+    2) Predict the next latent representation given the current latent + an action
+    3) Predict various targets (e.g. reward, value)
+
+    Amont the above, 1) can be used as the representation in comibination with
+    another RL aalgorithm; 2) and 3) can be used in policy improvements that
+    requires a predictive model (e.g. Monte Carlo Tree Search).
+
+    The model is trained with supervision on target prediction in 2) and 3).
+    Some of the targets may be computed with the reanalyze component. Please
+    refer to the original MuZero paper and the following paper for details.
+
+    `Online and Offline Reinforcement Learning by Planning with a Learned Model <https://arxiv.org/abs/2104.06294>`_.
+
     """
 
     def __init__(
             self,
             observation_spec,
             action_spec,
             model_ctor,
-            mcts_algorithm_ctor,
-            num_unroll_steps,
-            td_steps,
+            num_unroll_steps: int,
+            td_steps: int,
+            discount: float,
```

Comment:
it's kind of annoying that discount needs to be specified multiple times in MCTSAlgorithm and here.

Reply:
Right. As a mitigation here I put it in `MuzeroAlgorithm` and asked it to distribute `discount` to both `MCTSAlgorithm` and `MuzeroRepresentationImpl`. This is not the case for `Agent` though.